src/coreclr/src/vm/domainfile.cpp

```diff
@@ -2366,6 +2346,164 @@ DomainAssembly::EnumMemoryRegions(CLRDataEnumMemoryFlags flags)
     }
 }
 
-
 #endif // #ifdef DACCESS_COMPILE
 
+
+AssemblyNameIndex::AssemblyNameIndex(const SString& name, int32_t index)
+    : Name(name), Index(index)
+{
+    Name.Normalize();
+}
+
+#ifndef DACCESS_COMPILE
+
+DomainCompositeImage::DomainCompositeImage(
+    AppDomain *pDomain,
+    PEFile *peFile,
+    PEImage *peImage,
+    READYTORUN_HEADER *header,
+    LPCUTF8 compositeImageName,
+    uint8_t compositeImageNameLength,
+    LoaderAllocator *loaderAllocator,
+    AllocMemTracker& amTracker)
+  : DomainFile(pDomain, peFile, loaderAllocator)
+{
+    CONTRACTL
+    {
+        CONSTRUCTOR_CHECK;
+        STANDARD_VM_CHECK;
+        INJECT_FAULT(COMPlusThrowOM(););
+    }
+    CONTRACTL_END;
+
+    LoaderHeap *pHeap = loaderAllocator->GetHighFrequencyHeap();
+    m_utf8SimpleName = compositeImageName;
+    m_utf8SimpleNameLength = compositeImageNameLength;
+    m_peImage = peImage;
+    m_runEagerFixups = true;
+    
+    m_readyToRunInfo.Assign(new (amTracker.Track(pHeap->AllocMem((S_SIZE_T)sizeof(ReadyToRunInfo))))
+        ReadyToRunInfo(/*pModule*/ NULL, peImage->GetLoadedLayout(), header, /*compositeImage*/ NULL, &amTracker), /*takeOwnership*/ TRUE);
+    m_componentAssemblies = m_readyToRunInfo->GetComposite()->FindSection(ReadyToRunSectionType::ComponentAssemblies);
+    m_componentAssemblyCount = m_componentAssemblies->Size / sizeof(READYTORUN_COMPONENT_ASSEMBLIES_ENTRY);
+    
+    // Check if the current module's image has native manifest metadata, otherwise the current->GetNativeAssemblyImport() asserts.
+    m_manifestMetadata = peImage->GetNativeMDImport();
+
+    HENUMInternal assemblyEnum;
+    HRESULT hr = m_manifestMetadata->EnumAllInit(mdtAssemblyRef, &assemblyEnum);
+    mdAssemblyRef assemblyRef;
+    int assemblyIndex = 0;
+    while (m_manifestMetadata->EnumNext(&assemblyEnum, &assemblyRef))
+    {
+        LPCSTR assemblyName;
+        hr = m_manifestMetadata->GetAssemblyRefProps(assemblyRef, NULL, NULL, &assemblyName, NULL, NULL, NULL, NULL);
+        m_assemblySimpleNameToIndexMap.Add(new (amTracker.Track(pHeap->AllocMem((S_SIZE_T)sizeof(AssemblyNameIndex))))
+            AssemblyNameIndex(SString(SString::Ansi, assemblyName), assemblyIndex));
+        assemblyIndex++;
+    }
+}
+
+DomainCompositeImage::~DomainCompositeImage()
+{
+    CONTRACTL
+    {
+        DESTRUCTOR_CHECK;
+        NOTHROW;
+        GC_TRIGGERS;
+        MODE_ANY;
+        INJECT_FAULT(COMPlusThrowOM(););
+    }
+    CONTRACTL_END;
+}
+
+bool DomainCompositeImage::HasSimpleName(LPCUTF8 utf8SimpleName, uint32_t utf8Length) const
+{
+    return utf8Length == m_utf8SimpleNameLength && !memcmp(utf8SimpleName, m_utf8SimpleName, utf8Length);
+}
+
+bool DomainCompositeImage::TestAndClearRunEagerFixups()
```

Comment:
The intent of this function is difficult to decipher from the name. How about something like "NeedRunEagerFixups" or something along those lines?

Reply:
For now I renamed this to <code>EagerFixupsNeedToRun</code>.